contracts/network/BancorNetwork.sol

```diff
@@ -607,15 +606,35 @@ contract BancorNetwork is IBancorNetwork, Upgradeable, ReentrancyGuardUpgradeabl
         uint256 deadline,
         address beneficiary
     ) external payable whenNotPaused nonReentrant returns (uint256) {
-        _verifyTradeParams(sourceToken, targetToken, targetAmount, maxSourceAmount, deadline);
+        return _tradeByTargetAmount(sourceToken, targetToken, targetAmount, maxSourceAmount, deadline, beneficiary, msg.sender);
+    }
 
-        return
-            _trade(
-                TradeTokens({ sourceToken: sourceToken, targetToken: targetToken }),
-                TradeParams({ bySourceAmount: false, amount: targetAmount, limit: maxSourceAmount }),
-                TraderInfo({ trader: msg.sender, beneficiary: beneficiary }),
-                deadline
-            );
+    /**
+     * @inheritdoc IBancorNetwork
+     */
+    function tradeBySourceAmountArb(
+        Token sourceToken,
+        Token targetToken,
+        uint256 sourceAmount,
+        uint256 minReturnAmount,
+        uint256 deadline,
+        address beneficiary
+    ) external payable whenNotPaused only(_bancorArbitrage) returns (uint256) {
+        return _tradeBySourceAmount(sourceToken, targetToken, sourceAmount, minReturnAmount, deadline, beneficiary, msg.sender);
+    }
+
+    /**
+     * @inheritdoc IBancorNetwork
+     */
+    function tradeByTargetAmountArb(
+        Token sourceToken,
+        Token targetToken,
+        uint256 targetAmount,
+        uint256 maxSourceAmount,
+        uint256 deadline,
+        address beneficiary
+    ) external payable whenNotPaused only(_bancorArbitrage) returns (uint256) {
+        return _tradeByTargetAmount(sourceToken, targetToken, targetAmount, maxSourceAmount, deadline, beneficiary, msg.sender);
```

Comment:
For both `tradeBySourceAmountArb` and `tradeByTargetAmountArb`:

Can anyone here thinks of a scenario where `nonReentrant` is required because `only(_bancorArbitrage)` somehow doesn't prevent reentrancy?

Reply:
Actually, with the current design, if nonReentrant is added to this function, the arbs won't work - since the arb contract does a flashloan + trade, the V3 network gets locked after the flashloan - and adding `nonReentrant` here will revert the arb.
One important point to be considered is that multiple trades in one call will be possible from the arb contract with this change - something which wasn't possible before.